src/components/Box.jsx

```diff
@@ -26,11 +28,23 @@ const Title = styled.h2`
 `;
 
 const Box = () => {
+    const [index, setIndex] = useState(0);
+
+    const lastIndex = [...data].length - 1;
+    const currentSlide = [...data][index];
```

Comment:
Why do you recreated the array? You read the data so there is no mutation and no reasons to recreate some structures.